scripts/gwdd.py

```diff
@@ -0,0 +1,130 @@
+        self.shortname = "GWDD"
+        self.urls = {"GWDD": "http://datadryad.org/bitstream/handle/10255/dryad.235/GlobalWoodDensityDatabase.xls?sequence=1"}
+        self.tags = ["Taxon > Plants", "Spatial Scale > Global",
+                     "Data Type > Observational"]
+        self.ref = "http://datadryad.org/resource/doi:10.5061/dryad.234"
+        self.addendum = """Citation for the database: Zanne, A.E., Lopez-Gonzalez, G.*, Coomes, D.A., Ilic, J., Jansen, S., Lewis, S.L., Miller, R.B., Swenson, N.G., Wiemann, M.C., and Chave, J. 2009. Global wood density database. Dryad. Identifier: http://hdl.handle.net/10255/dryad.235. 
+        *Correspondence for updates to the database: G.Lopez-Gonzalez@leeds.ac.uk
+        For descriptions of the database, see Chave et al. 2009. Towards a worldwide wood economics spectrum. Ecology Letters. Identifier: http://hdl.handle.net/10255/dryad.234
+
+        Below we list the rules of use for the Global wood density database. 
+        These are developed based on the rules of use for the Glopnet dataset (www.nature.com/nature/journal/v428/n6985/full/nature02403.html) and Cedar Creek LTER and Related Data (http://www.lter.umn.edu/cgi-bin/register). 
+        If you would like to use the Global wood density database, we request that you:
+        1. Notify the main address of correspondence (Gaby Lopez-Gonzalo) if you plan to use the database in a publication.
+        2. Provide recognition of the efforts of this group in the assembly of the data by using the citation for the database above.
+        3. Recognize that these data were assembled by the group for various analyses and research questions. If any of these uses overlap with your interests, you recognize that group has precedence in addressing these questions."""
+
+    def download(self, engine=None, debug=False):
+        Script.download(self, engine, debug)
+
+        self.engine.download_file(self.urls["GWDD"], "GlobalWoodDensityDatabase.xls")
+        filename = os.path.basename("GlobalWoodDensityDatabase.xls")
+
+        book = xlrd.open_workbook(self.engine.format_filename(filename))
+        sh = book.sheet_by_index(1)
+        rows = sh.nrows
+
+        #Creating data table
+        lines = []
+        for i in range(1, rows):
+            row = sh.row(i)
+            if not all(Excel.empty_cell(cell) for cell in row):
+                try:
+                    this_line = {}
+
+                    def format_value(s):
+                        s = Excel.cell_value(s)
+                        return str(s).title().replace("\\", "/").replace('"', '')
+                    for num, label in enumerate(["Number", "Family", "Binomial", "Wood_Density",
+                                "Region", "Reference_Number"]):
+                        this_line[label] = format_value(row[num])
+                    lines.append(this_line)
+                except:
+                    raise
+                    pass
```

Comment:
What is the try/except accomplishing here? At a first glance it looks like if the try doesn't work it will just raise the exception anyway, so it may not be necessary.